### ISSUE TITLE
ephem.constellation accuracy

### ISSUE THREAD

**Issue by mattiaverga** [brandon-rhodes/pyephem#193]
I think the `constellation()` method isn't enough accurate when returning an object position.
For example:
```
>>> ogg = ephem.readdb('IC1101,f,15:10:56.10,+05:44:41.2,0,2000')
>>> ogg.compute()
>>> ephem.constellation(ogg)
('Se1', 'Serpens Caput')
>>> ogg.a_ra
3.974703755413636
>>> ogg.a_dec
0.10026528701762537
```

But looking at those coords in a planetarium software, IC1101 lays in Virgo boundaries (by less then 4 arcmins):
![immagine](https://user-images.githubusercontent.com/17218209/109547057-3ae57c80-7acb-11eb-9612-54f2fe601d59.png)

**Comment by brandon-rhodes** [comment]
Thanks for letting us know! Do you know what particular mistake PyEphem might be making in its code? I wonder if it’s careful to rotate the position into B1875.0 before comparing a position to the constellation boundaries.

**Comment by mattiaverga** [comment]
I see pyphem uses C code to compute and I'm not familiar with that.
However I suppose there's some rounding problem somewhere: it seems the error happens with a threshold of 15' (both in RA and Dec) which is quite a suspicious value. I mean, constellation boundaries coordinates seems to be 15' to the left in RA and 15' uppermost in Dec.

**Comment by brandon-rhodes** [comment]
I checked, and the code at least makes an attempt to do the correct precession; in the `constel.c` function `cns_pick()` I see:

```
   cal_mjd( 1, 1.0, 1875, &Mjd );
   precess( e, Mjd, &r, &d );
```
So unless the precession there is being somehow applied wrong, there must be some other problem. I probably won’t have time soon to do a full investigation to learn more, but thanks for opening the issue so that we can track this problem until someone has time to dig in and learn more about where the 15-minute error is coming from.

**Comment by brandon-rhodes** [comment]
Since I'm planning a new release soon, I've taken advantage of the holiday weekend here in the U.S. to dig into this issue. Turns out? The `constellation()` routine was using apparent RA/dec instead of apparent to try to decide the constellation! I've fixed it, so your new example — once the new version is released — should answer Virgo.

To install the development version in case you want to try out the fix ahead of time:
```
pip install -U https://github.com/brandon-rhodes/pyephem/archive/master.zip
```
